File: JAVASCRIPT/Easy/space-age.js
// Solution for Space Age

export const age = (planet, seconds) => {
  const earthYearInSecs = 31557600;
  const orbitTime = {
    Mercury: 0.2408467,
    Venus: 0.61519726,
    Earth: 1.0,
    Mars: 1.8808158,
    Jupiter: 11.862615,
    Saturn: 29.447498,
    Uranus: 84.016846,
    Neptune: 164.79132,
  };
  const properTime =
    planet.charAt(0).toUpperCase() + planet.slice(1).toLowerCase();
  const time = orbitTime[properTime];
  if (time === undefined) throw new Error(`Unknown planet: ${planet}`);
  const planetAge = seconds / (earthYearInSecs * time);
  return parseFloat(planetAge.toFixed(2));
};

File: C/Space Age/space_age.c
#include "space_age.h"
#include <stdio.h>
#define EARTH_YEAR_IN_SECONDS 31557600.0

static const float orbital_periods[] = {
    0.2408467,  // Mercury
    0.61519726, // Venus
    1.0,        // Earth
    1.8808158,  // Mars
    11.862615,  // Jupiter
    29.447498,  // Saturn
    84.016846,  // Uranus
    164.79132   // Neptune
};

float age(planet_t planet, int64_t seconds)
{
    if (planet < MERCURY || planet > NEPTUNE)
    {
        fprintf(stderr, "Error: Invalid planet\n");
        return -1.0;
    }
    float earth_years = seconds / EARTH_YEAR_IN_SECONDS;
    return earth_years / orbital_periods[planet];
}